modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala

```diff
@@ -279,6 +284,65 @@ object ObservationService {
           }
         }
 
+      def cloneObservation(
+        input: CloneObservationInput
+      ): F[Result[Observation.Id]] = 
+        session.transaction.use { xa =>
+
+          // We need the pid to do access checking, so let's just select it
+          val selPid = sql"select c_program_id from t_observation where c_observation_id = $observation_id"
+          session.prepareR(selPid.query(program_id)).use(_.option(input.observationId)).flatMap {
+
+            case None => Result.failure(s"No such observation: ${input.observationId}").pure[F]
+
+            case Some(pid) =>
+
+              // Ok the obs exists, so let's clone its main row in t_observation. If this returns
+              // None then it means the user doesn't have permission to see the obs.
+              val cObsStmt = Statements.cloneObservation(pid, input.observationId, user)
+              val cObs = session.prepareR(cObsStmt.fragment.query(observation_id)).use(_.option(cObsStmt.argument))
+              
+              // Ok let's do the clone
+              cObs.flatMap {
+
+                case None => 
+                  // User doesn't have permission to see the obs
+                  Result.failure(s"No such observation: ${input.observationId}").pure[F]
+
+                case Some(oid2) =>
+
+                  // The remaining clone operations yield no results and can't fail (logically, anyway)
+                  val cloneTheRest = 
+                    List(
+                    Statements.cloneAsterism(input.observationId, oid2),
+                    Statements.cloneGmosNorthStatic(input.observationId, oid2),
+                    Statements.cloneGmosNorthDynamic(input.observationId, oid2),
+                    Statements.cloneGmosNorthLongSlit(input.observationId, oid2),
+                  ).traverse_(af => session.prepareR(af.fragment.command).use(_.execute(af.argument)))
```

Comment:
How do we know it is GMOS North?  Where is GMOS South?

Reply:
I'm just copying rows without regard to what the actual mode is, but it would probably be more efficient to read in the mode and do something less dumb. And yes I supposed it should also support GS.